proposals/NNNN-android-manager-apis.md

```diff
@@ -0,0 +1,689 @@
+
+```java
+public abstract class BaseSubManager {
+
+	// allows better handling of sub managers by maintaining their states
+	public enum ManagerState {
+		SETTING_UP,
+		READY,
+		SHUTDOWN,
+		ERROR,;
+	}
+
+	protected ISdl internalInterface;
+	protected ManagerState state;
+
+	public BaseSubManager(@NonNull ISdl internalInterface){
+		this.internalInterface = internalInterface;
+		this.state = ManagerState.SETTING_UP;
+	}
+
+	/**
+	 * <p>Called when manager is being torn down</p>
+	 */
+	public void dispose(){
+		this.state = ManagerState.SHUTDOWN;
+	}
+
+	protected void transitionToState(ManagerState state) {
+		this.state = state;
+	}
+
+	protected ManagerState getState() {
+		return state;
+	}
+}
+	
+```
+
+#### LockScreenManager (and default lock screen)
+
+The lockscreen manager maintains the base logic as the previous lockscreen manager, but also adds in key functionality that was missing previously. New items include a default lockscreen (layout and activity) and the ability to customize that default lockscreen layout. Apps are also able to pass in custom layouts or disable it completely and roll their own.
+
+```java
+public class LockscreenManager extends BaseSubManager {
+
+    public interface OnLockScreenIconDownloadedListener{
+        public void onLockScreenIconDownloaded(Bitmap icon);
+        public void onLockScreenIconDownloadError(Exception e);
+    }
+
+    public LockscreenManager(LockScreenConfig lockScreenConfig, Context context, ISdl internalInterface){}
+
+    public Bitmap getLockScreenIcon(){}
+
+}
+```
+##### Lockscreen Config
+
+```java
+public class LockScreenConfig {
+
+	public LockScreenConfig(){}
+
+	public void setEnabled(Boolean enableAutomaticLockScreen){}
+
+	public Boolean getEnabled() {}
+
+	public void setBackgroundColor(Color backgroundColor){}
+
+	public Color getBackgroundColor() {}
+
+	public void setAppIcon(int appIconInt) {}
+
+	public int getAppIcon() {}
+	
+	public void setCustomView(int customLayoutResource) {}
+
+	public int getCustomView() {}
+}
+```
+
+The lockscreen manager will instantiate and manage the lockscreen activity and set the parameters appropriately depending on the lockscreen configuration provided by the developer. If `setEnabled` is set to `False`, the developer will need to manage their lockscreen manually. 
+
+#### ScreenManager
+
+This manager closely follows to the previously accepted and implemented iOS manager proposal for [Show Manager](https://github.com/smartdevicelink/sdl_evolution/pull/268), also known as the Screen Manager. It gives the developer the ability to easily manage the items on the screen without having to worry about things such as the special ordering of RPCs needed and reduces a large amount of boiler plate logic.
```

Comment:
First sentence should be "...closely follows the previously accepted..." and reference the proposal file (https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0134-ios-show-manager.md)